### PR TITLE
SNOW-532834 Using default role when snowflake.role.name is not specified

### PR DESCRIPTION
**Pull request by sfc-gh-alhuang**
Use default roles when user does not specified `role` properties in [Streaming Ingest Client SDK](https://github.com/snowflakedb/snowflake-ingest-java).
[Related issue](https://github.com/snowflakedb/snowflake-kafka-connector/issues/571#issuecomment-1468495253)

[SNOW-532834](https://snowflakecomputing.atlassian.net/browse/SNOW-532834)

[Server side PR](https://github.com/snowflakedb/snowflake/pull/103811)

[SNOW-532834]: https://snowflakecomputing.atlassian.net/browse/SNOW-532834?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/utils/Utils.java [hunk 1]
```diff
@@ -131,10 +131,6 @@ public static Properties createProperties(Properties inputProp) {
               Integer.parseInt(properties.get(Constants.PORT).toString())));
     }
 
-    if (!properties.containsKey(Constants.ROLE)) {
-      throw new SFException(ErrorCode.MISSING_CONFIG, "role");
```
Comment (sfc-gh-xhuang) [inline src/main/java/net/snowflake/ingest/utils/Utils.java:135]: should we throw a logInfo to indicate that the default role is being used?

Comment (sfc-gh-alhuang) [reply src/main/java/net/snowflake/ingest/utils/Utils.java:135]: Good to have one, added [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/0b2c10b0466c2d0f01424faedbe3f74f4e0993cb/src/main/java/net/snowflake/ingest/utils/Utils.java#L134).

**Review by sfc-gh-rcheng — CHANGES_REQUESTED** [review]
your format gate is failing, run the formatter (format.sh) from root to fix it. I ended up creating an alias to format, git commit, git push to auto format for me.

you can ignore the cla bot for now, it recently kind of flakey unfortunately

File: src/main/java/net/snowflake/ingest/utils/Utils.java (hunk 1 above)
Comment (sfc-gh-alhuang) [reply src/main/java/net/snowflake/ingest/utils/Utils.java:135]: Fixed.

**Review by sfc-gh-xhuang — APPROVED** [review]
looks good to me.
 I think you need the server side changed merged first before you can merge this to the next release?

File: src/main/java/net/snowflake/ingest/utils/Utils.java [hunk 2]
```diff
@@ -132,7 +132,9 @@ public static Properties createProperties(Properties inputProp) {
     }
 
     if (!properties.containsKey(Constants.ROLE)) {
-      throw new SFException(ErrorCode.MISSING_CONFIG, "role");
+      logger.logInfo(
+          "Missing user role property in the configuration file, the default user role will be"
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/utils/Utils.java:136]:
```suggestion
          "Snowflake role is not provided, user's default role will be used"
```

It's not always we use configuration file (For making our lives easier, testing uses this profile.json)

**Review by sfc-gh-japatel — COMMENTED** [review]
thanks @sfc-gh-alhuang how have you tested this?

**Comment by sfc-gh-alhuang** [comment]
> thanks @sfc-gh-alhuang how have you tested this?

@sfc-gh-japatel I've tested this via some SDK function calls. For the server side, I've modified some regression tests and add some unit tests.

**Review by sfc-gh-rcheng — APPROVED** [review]

**Review by sfc-gh-tzhang — APPROVED** [review]
lgtm, ship it! Could you work with @sfc-gh-xhuang and Lei to update our documentation?

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm.

**Pull request merged**